tests/integration/acp/p2p/replicator_with_doc_actor_relationship_test.go

```diff
@@ -0,0 +1,219 @@
+
+package test_acp_p2p
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/sourcenetwork/immutable"
+
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+)
+
+func TestACP_P2PReplicatorWithPermissionedCollectionCreateDocActorRelationship_SourceHubACP(t *testing.T) {
+	expectedPolicyID := "fc56b7509c20ac8ce682b3b9b4fdaad868a9c70dda6ec16720298be64f16e9a4"
+
+	test := testUtils.TestCase{
+
+		Description: "Test acp, p2p replicator with collection that has a policy, create a new doc-actor relationship",
+
+		SupportedACPTypes: immutable.Some(
+			[]testUtils.ACPType{
+				testUtils.SourceHubACPType,
+			},
+		),
+
+		Actions: []any{
+			testUtils.RandomNetworkingConfig(),
+
+			testUtils.RandomNetworkingConfig(),
+
+			testUtils.AddPolicy{
+
+				Identity: immutable.Some(1),
+
+				Policy: `
+                    name: Test Policy
+
+                    description: A Policy
+
+                    actor:
+                      name: actor
+
+                    resources:
+                      users:
+                        permissions:
+                          read:
+                            expr: owner + reader + writer
+
+                          write:
+                            expr: owner + writer
+
+                          nothing:
+                            expr: dummy
+
+                        relations:
+                          owner:
+                            types:
+                              - actor
+
+                          reader:
+                            types:
+                              - actor
+
+                          writer:
+                            types:
+                              - actor
+
+                          admin:
+                            manages:
+                              - reader
+                            types:
+                              - actor
+
+                          dummy:
+                            types:
+                              - actor
+                `,
+
+				ExpectedPolicyID: expectedPolicyID,
+			},
+
+			testUtils.SchemaUpdate{
+				Schema: fmt.Sprintf(`
+						type Users @policy(
+							id: "%s",
+							resource: "users"
+						) {
+							name: String
+							age: Int
+						}
+					`,
+					expectedPolicyID,
+				),
+			},
+
+			testUtils.ConfigureReplicator{
+				SourceNodeID: 0,
+
+				TargetNodeID: 1,
+			},
+
+			testUtils.CreateDoc{
+				Identity: immutable.Some(1),
+
+				NodeID: immutable.Some(0),
+
+				CollectionID: 0,
+
+				DocMap: map[string]any{
+					"name": "Shahzad",
+				},
+			},
+
+			testUtils.WaitForSync{},
+
+			testUtils.Request{
+				// Ensure that the document is hidden on all nodes to an unauthorized actor
+				Identity: immutable.Some(2),
+
+				Request: `
+					query {
+						Users {
+							name
+						}
+					}
+				`,
+
+				Results: map[string]any{
+					"Users": []map[string]any{},
+				},
+			},
+
+			testUtils.AddDocActorRelationship{
+				NodeID: immutable.Some(0),
+
```

Comment:
question: I've seen you do this quite a bit, why do you add whitespace between every property in an instance declaration?

Reply:
No good reason, I am claustrophobic perhaps and like spaced lines lol. I often wonder why people don't add whitespaces.

Reply:
TBH it's just a habit while writing code, I don't care if some formatter formats these style choices to one format that is more consistent (idk if there is one, or a linter rule?)